lib/src/rules/avoid_returning_this.dart

```diff
@@ -63,6 +61,29 @@ class AvoidReturningThis extends LintRule {
   }
 }
 
+class _BodyVisitor extends UnifyingAstVisitor {
```

Comment:
I believe that this could be a subclass of `RecursiveAstVisitor`. The difference is that a `UnifyingAstVisitor` will invoke `visitNode` on every node in addition to the normal `visit` method, so there's an extra method invocation per node. Probably won't impact performance by a measurable amount, but it will also keep the code a bit cleaner.

Reply:
Great catch.  Thanks!